tikv-client/src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java

```diff
@@ -127,163 +128,194 @@ public void resolve() {
     typeMap = inferrer.getTypeMap();
   }
 
-  public DAGRequest buildScan(boolean idxScan) {
-    if (idxScan) {
-      return buildIndexScan();
-    } else {
-      return buildTableScan();
-    }
+  private static ColumnInfo handleColumn() {
+    return ColumnInfo.newBuilder()
+        .setColumnId(-1)
+        .setPkHandle(true)
+        // We haven't changed the field name in protobuf file, but
+        // we need to set this to true in order to retrieve the handle,
+        // so the name 'setPkHandle' may sounds strange.
+        .build();
   }
 
-  private DAGRequest buildIndexScan() {
+  /**
+   * merge buildIndexScan and buildTableScan since they are very much alike.
```

Comment:
I prefer `unify` to `merge` in this comment, and method `buildIndexScan` and `buildTableScan` are nowhere seen here, direct reference to them might brought some confuse. I recommend we comment as `Unify index scan and table scan building logic here since they are very much alike.`

Reply:
aggree